modules/util/util.go

```diff
@@ -66,40 +68,59 @@ func IsEmptyString(s string) bool {
 	return len(strings.TrimSpace(s)) == 0
 }
 
-// NormalizeEOL will convert Windows (CRLF) and Mac (CR) EOLs to UNIX (LF)
-func NormalizeEOL(input []byte) []byte {
-	var right, left, pos int
-	if right = bytes.IndexByte(input, '\r'); right == -1 {
-		return input
+type normalizeEOLReader struct {
+	rd           io.Reader
+	isLastReturn bool
+}
+
+func (r *normalizeEOLReader) Read(bs []byte) (int, error) {
+	var p = make([]byte, len(bs))
+	n, err := r.rd.Read(p)
+	if err != nil {
+		return n, err
 	}
-	length := len(input)
-	tmp := make([]byte, length)
-
-	// We know that left < length because otherwise right would be -1 from IndexByte.
-	copy(tmp[pos:pos+right], input[left:left+right])
-	pos += right
-	tmp[pos] = '\n'
-	left += right + 1
-	pos++
-
-	for left < length {
-		if input[left] == '\n' {
-			left++
-		}
 
-		right = bytes.IndexByte(input[left:], '\r')
-		if right == -1 {
-			copy(tmp[pos:], input[left:])
-			pos += length - left
-			break
+	var j = 0
+	for i, c := range p[:n] {
+		if i == 0 {
+			if c == '\n' && r.isLastReturn {
+				r.isLastReturn = false
+				continue
+			}
+			r.isLastReturn = false
+		}
+		if c == '\r' {
+			if i < n-1 {
+				if p[i+1] != '\n' {
+					bs[j] = '\n'
+				} else {
+					continue
+				}
+			} else {
+				r.isLastReturn = true
+				bs[j] = '\n'
+			}
+		} else {
+			bs[j] = c
 		}
-		copy(tmp[pos:pos+right], input[left:left+right])
-		pos += right
-		tmp[pos] = '\n'
-		left += right + 1
-		pos++
+		j++
 	}
-	return tmp[:pos]
+
+	return j, nil
+}
```

Comment:
This might be a significant performance degrade from the previous implementation. IndexByte(...) is - somewhat counterintuitively - much faster than iterating through the byte slice.

We may be better off having a small buffer so that the indexByte calls are reduced.

Reply:
Please send another PR as I think it's not related with this PR.

Reply:
I have revert these changes.